File: 7/27/3.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    string answer = "";

    string upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string lower = "abcdefghijklmnopqrstuvwxyz";

    int tmp = 0;


    for (int i = 0; i < s.length(); i++){

        if(s[i] == ' '){
            tmp = 0;
            continue;

        }

        if(tmp == 0){
            if(s[i]>=97)
                s[i] = upper[(s[i]-97)%26];
            tmp = 1;
        }

        else{
            if(s[i]>=65 && s[i] <=90){
                s[i] = lower[(s[i]-65)%26];
            tmp = 0;
        }

      

    }


    answer = s;
    return answer;
}

File: 2021/5/8/3.py
from _collections import deque


def solution(n, k, cmd):
    answer = ''

    arr = [True for _ in range(n)]
    nexts = [i+1 for i in range(n)]
    befores = [i-1 for i in range(n)]
    curIdx = k
    delIndices = deque()
    for cm in cmd :
        c = cm.split()

        if c[0] == "D" :
            for _ in range(int(c[1])) :
                curIdx = nexts[curIdx]

        elif c[0] == "U" :
            for _ in range(int(c[1])):
                curIdx = befores[curIdx]


        elif c[0] == "C":
            arr[curIdx] = False


            if befores[curIdx] != 0 :
                nexts[befores[curIdx]] = nexts[curIdx]

            if nexts[curIdx] != n :
                befores[nexts[curIdx]] = befores[curIdx]

            delIndices.append(curIdx)

            if nexts[curIdx] >= n :
                curIdx = befores[curIdx]
            else :
                curIdx = nexts[curIdx]

        else :
            tmp = delIndices.pop()
            arr[tmp] = True
            if befores[tmp] != -1 :
                nexts[befores[tmp]] = tmp
            if nexts[tmp] < n :
                befores[nexts[tmp]] = tmp

    for i in range(n) :
        if arr[i] :
            answer += "O"
        else :
            answer += "X"

    return answer

print(solution(8,2,["D 2","C","U 3","C","D 4","C","U 2","Z","Z"]))